NS-008: Updates to CA Infrastructure Scope, Trusted Roles, Systems' Applicability, and various other improvements

clintwilson opened the pull request:
Ballot NS-008 is proposed by Clint Wilson (Apple) and endorsed by Ben Wilson (Mozilla), Cade Cairns (Google Trust Services), and Daniel Jeffery (Fastly).

## Overview
### Definition Changes
#### Added Definitions

* Network Boundary Control: New definition for components that manage network traffic flow; replaces "Network Equipment"
* Principle of Separation of Duties: Added definition describing how tasks are divided among multiple individuals

#### Modified Definitions

* CA Infrastructure: Simplified to reference only "Certificate System", "Root CA System", and "Security Support System" (removed references to "Certificate Management System", "Delegated Third Party System", and "Issuing System")
* Certificate System: Modified definition to include "store" and "access" in activities list
* Critical Security Event: Restructured with numbered lists and expanded examples
* Security Support System: Changed from a "system or set of systems" to "The System(s)" and removed "network boundary control" from the list of functions
* Trusted Role: Changed from "employee or contractor" to "individual employee or contractor" and reduced scope to those with access to a "Certificate System or Root CA System"

#### Removed Definitions

* Certificate Management System definition has been removed
* Delegated Third Party System definition has been removed
* Issuing System definition has been removed

### Structural and Terminology Changes

* Network Equipment → Network Boundary Controls: Throughout the document, references to "Network Equipment" have been replaced with "Network Boundary Controls"
* Inventory requirement moved: Requirement moved from Section 4 to beginning of Section 1: "The CA MUST define an inventory of its CA Infrastructure"
* Section 4 restructuring:
  * Section number changes: 4.3 → 4.2, 4.4 → 4.3
  * Former 4.2 "Intrusion Detection and Prevention" moved to 4.1

### Substantive Requirement Changes
#### Section 1: CA Infrastructure and Network Configuration

* Root CA Systems isolation requirement: Added explicit requirement in section 1.2.1: "Root CA Systems MUST be on physically separate networks from all other CA Infrastructure"
* Network segmentation implementation: Section 1.1.1.1 and 1.1.1.2 have switched places:
  * Network segmentation design principles moved to 1.1.1.1 (previously in 1.1.1.2)
  * Implementation details moved to 1.1.1.2 (previously in 1.1.1.1)
  * "Software-defined networking" moved from MAY to MUST category

#### Section 2: Access Control

* Trusted Roles scope: Changed from designing/building/maintaining "CA Infrastructure and Network Equipment" to specifically "Certificate Systems and Root CA Systems"
* Separation of Duties: Replaced requirement that Trusted Roles be assigned consistent with requirements of Multi-Party Control with a requirement that Trusted Roles be assigned consistent with "the Principle of Separation of Duties" (section 2.1)
* Access restriction scope: Changed from "CA Infrastructure and/or Network Equipment" to "Certificate Systems and Root CA Systems" in section 2.2.1 and 2.2.1.1
* Multi-Factor Authentication: Modified section 2.2.3 to require MFA "for access to CA Infrastructure" rather than "for accounts on CA Infrastructure and access to CA Infrastructure"

#### Section 4: Vulnerability Management

* Expanded scope for vulnerability policies: 
  * Added Security Support Systems and Network Boundary Controls to recommended scope
  * Added New deadline: "Effective 15-Apr-2026, these policies and procedures MUST apply to Security Support Systems and Network Boundary Controls"

#### Date Format Changes

 * Date format standardized to DD-MMM-YYYY throughout

---

Following discussions in the NSWG, this is an example version of the NCSSRs which removes several of the "Systems" from the scope of CA Infrastructure (and the document in general). The changes are based on the assumed completion of IPR review for NS-004 and NS-006.

As also touched upon in NSWG meeting on Nov 19, 2024, I slightly updated the Trusted Role definition to (hopefully) avoid its interpretation as including companies not involved in the operation of CA Infrastructure.

Additionally, fixed some minor issues arising from NS-004 and NS-006:
* Reversion of Workstation definition
* Orphaned text in 1.2.2
* Markdown formatting in 2.2.1.2

### Potential to-dos
* switch 1.1.1.1 and 1.1.1.2 ordering ✅
* update defined term used in Section 1 to CA Infrastructure ✅
  * should it include Network Equipment as well ❌ (from f2f discuss)
* make SDN in 1.1.1.1 a MUST ✅
* require physically separate networks for Root CA Systems and 3rd Party ✅ 
* move asset inventory and risk assessment portions of NCSSRs to first part of document 📆 
* MFA applicability in 2.2.3 may be overly broad
  * should it include both "accounts" and "access" (remove accounts from f2f discuss) ✅
* Replace MPC in 2.1 with "separation of duties" ✅ 
 * remove critical, over any individual sensitive task
* consistency of Trusted Roles (Certificate Systems and Root CA Systems) throughout section 2 (CA Infrastructure and Network Equipment) ✅ 
* acknowledge that 3.2 doesn't include Network Equipment (3.1 does) (neither does the definition of Critical Security Event); maybe fix in v2.1 📆 
* update formatting of Critical Security Event (i.e. make into list, fix 'or', etc.) ✅ 
* use of "and/or" vs "or" vs "and" in Section 2, review and ensure it makes sense ✅ 
* Review whether 3.c (root ca system air-gapped) should come back after changes in this ballot ❌ 
* 2.2.1.3 could clarify which authentication credentials are in scope (nice to have) 📆 
  * formatting bug in 2.2.1.2 ✅ 
* Consider extending 1.2.2 to include Network Equipment (at least in the SHOULD for within CA infra; maybe for a later version) ✅ 
* Fix risk assessment capitalization in 4.4 ✅ 
* Security Support Systems definition can't be removed yet 📆 
 * Need to add language to ensure authentication, authorization, and auditing are considered as central to the scope of the requirements as other components of the CA's infrastructure ❓ 
 * Access control systems (and other parts of Security Support Systems) should be in-scope of Section 4's vulnerability management ✅ <- added future effective date
* Update physically secure environment definition, "assessed in accordance with Risk Assessments" or something, maybe get more input from Dustin, look at mpic/open-mpic ✅
* remove network boundary control from security support system ✅
* add network boundary control to 4.1 since gone from security support system ✅

clintwilson committed:
CA Infrastructure Scope update

Following discussions in the NSWG, this is an example version of the NCSSRs which removes several of the "Systems" from the scope of CA Infrastructure (and the document in general).
The changes are based on the assumed completion of IPR review for NS-004 and NS-006.

Additionally, fixed some minor issues arising from NS-004 and NS-006:
* Reversion of Workstation definition
* Orphaned text in 1.2.2
* Markdown formatting in 2.2.1.2

romanf commented on docs/NSR.md:
What's the idea behind the "individual" here? At least for me as a non-native speaker "an employee" and "an individual employee" sound identical...

clintwilson replied:
I added this in to address confusion expressed around whether an entire company, used by a CA, could be considered a Trusted Role.

clintwilson commented on docs/NSR.md:
```diff
@@ -64,16 +64,10 @@ The following are outcomes that this document seeks to achieve:
-**Certificate Management System**: A system used by a CA or Delegated Third Party to process, approve issuance of, or store certificates or certificate status information, including the database, database server, and storage.
-
-**Certificate System**: A system used by a CA or Delegated Third Party to access, process, or manage data or provide services related to:
+**Certificate System**: A system used by a CA or Delegated Third Party to access, process, or manage data or provide services related to performing:
```
> to store, access, process, or manage data

miguelantonios replied:
Cloud Services Subgroup Concern: Removal of Support Systems from Certificate Systems no longer encompasses those periphery systems in any kind of requirements for controls. How can we incorporate those requirements into this definition?

clintwilson commented on docs/NSR.md:
```diff
@@ -445,37 +413,69 @@ The CA SHOULD ensure incident response plans minimally include:
+
+These policies and procedures MUST apply to all Certificate Systems.
+
+These policies and procedures SHOULD apply to Security Support Systems.
```
Security Support Systems has been removed as a defined term.

miguelantonios replied:
+1

BenWilson-Mozilla commented on docs/NSR.md:
```diff
@@ -219,7 +184,9 @@ CA Infrastructure MUST be in a Physically Secure Environment.
 ##### 1.2.2
 
-Connections to and within the CA Infrastructure MUST be authenticated and encrypted except OCSP and CRL.
+Connections to the CA Infrastructure MUST be authenticated and encrypted, except where documented that a formal specification prohibits or limits the use of authentication and/or encryption.
```
Maybe this should say ", except where a formal specification prohibits ..."?  It seems that "where documented that" adds an unnecessary record-creating requirement and could be deleted without changing meaning. But I'm open to other suggestions/comments.

BenWilson-Mozilla commented on docs/NSR.md:
For this objective, "CAs are able to clearly understand the minimum security requirements found in this document and successfully adapt/implement these Requirements to their own infrastructure and architecture" maybe add something like ", which requires a comprehensive inventory of their Certificate Systems to ensure that their infrastructure is adequately protected."

miguelantonios commented on docs/NSR.md:
```diff
@@ -445,37 +413,69 @@ The CA SHOULD ensure incident response plans minimally include:
+
+#### 4.1 Inventory of Certificate Systems
+
+The CA MUST define an inventory of Certificate Systems.
```
CS group feedback: what if we move this up so that it pertains not just to vuln management, which I don't believe was the intention?

miguelantonios commented on docs/NSR.md:
CS group thoughts: What if we use something similar to what is in section 3.1.2.1 "The CA SHOULD ensure retained and/or archived audit logs are kept and managed in a manner sufficiently effective to prevent unapproved alteration or access." for all other relevant security support system sections (e.g. change  management, vuln management, etc.) The benefit of this is that it would bring back the requirements to protect security support systems without blowing up the trusted roles bit (which was a concern in the past). 

Vuln management example: "Vuln management systems would have to be configured/locked down in a way that prevents malicious attacks on Certificate Systems"

miguelantonios reviewed (commented):
Leaving some comments from Cloud Services Subgroup meeting today. 

tl;dr: the concern that support security systems is removed and no longer requires controls on those systems. One possible solution is to hardcode those requirements and systems within their applicable sections (e.g. Access Management systems in section 2) similar to what's already done in 3.1.2.1 "The CA SHOULD ensure retained and/or archived audit logs are kept and managed in a manner sufficiently effective to prevent unapproved alteration or access."

clintwilson commented on docs/NSR.md:
```diff
@@ -82,21 +76,13 @@ The following are outcomes that this document seeks to achieve:
    7. certificate revocation;
-   8. authoritative certificate status; or
+   8. generation and signing of authoritative certificate status; or
    9. key escrow.
```
It's intended to apply also to escrow of Subscriber keys, where allowed (such as with S/MIME Certificates), so I believe it's correct as-is. Where key escrow is not performed by a CA, then this becomes not applicable (therefore neither constraining nor expanding the requirements applied to the CA).

clintwilson committed:
Update from comments

* Incorporated suggestions from several comments
* Moved Requirements of 4.1 to 1 (and relabeled 4 as 4.1 so as to not renumber all of 4's subsections)
* Attempted to address concerns with change in CA Infrastructure scope by:
  * re-adding (slightly modified) Security Support System definition
  * changing Trusted Role to apply to those with Certificate Systems or Root CA Systems access

clintwilson committed:
Merge branch 'main' into ca-infrastructure-scope

aww-aww commented on docs/NSR.md:
```diff
@@ -208,13 +201,10 @@ CA Infrastructure MUST be in a Physically Secure Environment.
 ##### 1.2.2
 
-Connections to the CA Infrastructure MUST be authenticated and encrypted, except where documented that a formal specification prohibits or limits the use of authentication and/or encryption.
+Connections to the CA Infrastructure MUST be authenticated and encrypted, except where formal specification(s) prohibits or limits the use of authentication and/or encryption.
```
The classic style guides (AP, Chicago) are split on how to handle parenthetical plurals, so there is not a clear right or wrong. Regardless, this reads oddly to me. It either needs s/where/where a/ or s/prohibits or limits/prohibit or limit/

clintwilson committed:
Incorporate NSWG Discussions

Based on discussions in NSWG meetings over the last couple months, changes have been made to:

* switch 1.1.1.1 and 1.1.1.2 ordering ✅
* update defined term used in Section 1 to CA Infrastructure ✅
* make SDN in 1.1.1.1 a MUST ✅
* require physically separate networks for Root CA Systems and 3rd Party ✅
* Replace MPC in 2.1 with "separation of duties" ✅
* consistency of Trusted Roles (Certificate Systems and Root CA Systems) throughout section 2 (CA Infrastructure and Network Equipment) ✅
* update formatting of Critical Security Event (i.e. make into list, fix 'or', etc.) ✅
* use of "and/or" vs "or" vs "and" in Section 2, review and ensure it makes sense ✅
* formatting bug in 2.2.1.2 ✅
* Consider extending 1.2.2 to include Network Equipment (at least in the SHOULD for within CA infra; maybe for a later version) ✅
* Fix risk assessment capitalization in 4.4 ✅
* Access control systems (and other parts of Security Support Systems) should be in-scope of Section 4's vulnerability management ✅ <- added future effective date

clintwilson committed:
Formatting

Fix a few instances of formatting inconsistency

clintwilson committed:
Merge branch 'main' into ca-infrastructure-scope

dzacharo commented:
Can someone please fix the headings? I think the entire document can be reduced by 1 level

The document starts with:

`## Introduction`

instead of:

`# Introduction`

clintwilson commented:
@dzacharo 
> Can someone please fix the headings? I think the entire document can be reduced by 1 level
> 
> The document starts with:
> 
> `## Introduction`
> 
> instead of:
> 
> `# Introduction`

We've replaced what would typically be the H1 (#) header with the Front Matter Title (table at the top) of CA/BF documents. H2 (##) is correct for the first section after the Front Matter, as far as I understand.

clintwilson committed:
F2F Discussion updates

Based on discussion at CA/BF F2F 64:
* updated definition of Physically Secure Environment
* removed "network boundary control" from definition of Security Support System (now its own definition)
* removed "accounts within CA Infrastructure" from Multi-Factor Authentication requirements of 2.2.3
* added Network Boundary Control to scope of Section 4's SHOULD and futre MUST requirements

clintwilson committed:
Update workflow file

Based on https://github.com/actions/runner-images/issues/11101, updating the build-draft-docs.yml to use ubuntu-24.04 instead of ubuntu-20.04

clintwilson commented on docs/NSR.md:
```diff
@@ -182,43 +190,44 @@ CA Infrastructure MUST be segmented into separate networks based on the function
-##### 1.2.2
+Root CA Systems MUST be on physically separate networks from all other CA Infrastructure.
+
+CA Infrastructure operated by a Delegated Third Party MUST be on physically separate networks from all other CA Infrastructure.
```
Remove

clintwilson commented on docs/NSR.md:
```diff
@@ -283,47 +293,51 @@ The CA MUST ensure personnel assigned to a Trusted Role act only within the scop
 ##### 2.2.1
 
-The CA MUST ensure access to CA Infrastructure and/or Network Equipment is:
+The CA MUST ensure access to CA Infrastructure and Network Boundary Controls is:
```
Certificate Systems and Root CA Systems

clintwilson commented on docs/NSR.md:
```diff
@@ -266,14 +275,15 @@ Within this Section 2, references to "access" include all physical and logical a
-   2. requirements of Multi-Party Control.
+1. the Principle of Least Privilege;
+2. the Principle of Separation of Duties; and
+3. any applicable requirements for Multi-Party Control.
```
Remove

clintwilson committed:
NSWG April 8, 2025 Updates

Based on discussion, updated 3 areas

aww-aww commented on docs/NSR.md:
```diff
@@ -49,7 +49,7 @@ The following are outcomes that this document seeks to achieve:
 | 1.2 | SC3 | Two-Factor Authentication and Password Improvements | 16‐Aug‐18 | 15-Sep-18 |
 | 1.3 | SC21 | The Network and Certificate Systems Security Requirements Section 3 (Log Integrity Controls) | 26‐Sep‐19 | 4-Nov-2019 |
-| 1.4 | SC29 | System Configuration Management | 7-May-20 | 8-Jun-2020 |
+| 1.4 | SC29 | System Configuration Management | 7-May-2020 | 8-Jun-2020 |
```
Nit: preceding entries for days are two digits, s/7/07/ ?

aww-aww commented on docs/NSR.md:
```diff
@@ -283,47 +290,51 @@ The CA MUST ensure personnel assigned to a Trusted Role act only within the scop
-
-   1. accounts on CA Infrastructure; and
-   2. access to CA Infrastructure.
+The CA MUST enforce the use of Multi-Factor Authentication for access to CA Infrastructure.
```
s/CA Infrastructure/Certificate Systems/ ?

cairnsc commented on docs/NSR.md:
```diff
@@ -444,19 +455,19 @@ The CA SHOULD ensure incident response plans minimally include:
 ### 4. Vulnerability Management
 
-The CA MUST implement the policies and procedures in this Section for identifying, evaluating, and resolving security vulnerabilities.
+#### 4.1
```
Nit from the cloud subgroup: this subsection seems out of place since it doesn't have a title. It also seems the requirements in this subsection are broadly applicable to section 4. Would it make sense to remoev this as a subsection and renumber the ones below?

clintwilson committed:
Address 2 comments

Update Document History Adopted and Effective Dates to be consistently formatted
Update Section 4, shifting 4.1 to 4, and 4.2 - 4.4 to 4.1 - 4.3 respectively.

clintwilson committed:
Fix link

clintwilson committed:
Remove critical from Separation of Duties

clintwilson committed:
Push effective date to November

and fix spelling error in Risk Assessment definition

clintwilson committed:
Address 2.2.6 feedback

A concern was raised related to the current wording of "remote connection[s] that enable.... access to CA Infrastructure" as allowing an interpretation to conclude that a customer accessing a CA's web portal may qualify as meeting the description, thus necessitating the customer be brought into scope of the bulleted requirements. This commit aims to remove the possibility of this interpretation by introducing a more precise term ("Privileged Access") for the type of access enabled in Section 2.2.6 which then necessitates the behavior defined therein.
This term is also used in the definition of "Workstation", as it was identified by the group as very applicable in that context.

Separately, I'll note that while a customer portal may be a system operated by a CA, and it's very likely to be an application system which *interfaces* with CA Infrastructure as defined in the NCSSRs, it's not inherently (and likely should _not_ be) the infrastructure itself. That application layer functions at a fundamentally different layer than the management and administration layers underlying the CA Infrastructure and Network Boundary Controls these requirements seek to ensure the security of. Hopefully this change makes that more clear, though that separation is important enough that it shouldn't hinge solely upon this change.

clintwilson committed:
Add Document Version

Add Document Version, Date, and Change History entry

clintwilson committed:
Post-ballot adoption and IPR Review Adminstrative Edits

Update dates following ballot voting and IPR review
